firmware/powerboard2/code/app/CAN_componentSpecific.c

```diff
@@ -0,0 +1,45 @@
+/*
+ * CAN_componentSpecific.c
+ *
+ *  Created on: Aug 8, 2020
+ *      Author: Poornachander
+ */
+
+#include "CAN.h"
+#include "messageHandler.h"
+
+static void CAN_enablePeripheralsClockCallback(void);
+static void CAN_messageReceivedCallback(const protocol_MID_E messageID, const protocol_allMessages_U * const message);
+
+static const CAN_HWConfig_S CAN_HWConfig =
+{
+	// GPIO
+	.rxPin = 8,
+	.txPin = 9,
+	.GPIOPort = GPIOB,
+	.AFNumber = GPIO_AF8_CAN1,
+
+	// CAN
+	.CANPeriph = CAN1,
+
+	// Common
+	.enablePeripheralsClockCallback = CAN_enablePeripheralsClockCallback,
+};
+
+extern const CAN_config_S CAN_config;
+const CAN_config_S CAN_config =
+{
+	.HWConfig = &CAN_HWConfig,
+	.messageReceivedCallback = CAN_messageReceivedCallback,
+};
+
+static void CAN_enablePeripheralsClockCallback(void)
+{
+	RCC_AHB1PeriphClockCmd(RCC_AHB1ENR_GPIOBEN, ENABLE);

```

Comment:
Instead of HAL functions, maybe switch it to StdLink. I dont like HAL, just a personal preference.

Reply:
This isn't HAL, it's StdPeriph, but I get what you're saying. I think you mean the CMSIS library style of writing to registers. I'd like to move towards this. This is much safer and less prone to mistakes. However, everyone should understand what's happening behind this function.